exporters/prometheus/src/exporter_utils.cc

```diff
@@ -242,40 +251,63 @@ void PrometheusExporterUtils::SetData(std::vector<T> values,
  * Histogram => Prometheus Histogram
  */
 template <typename T>
-void PrometheusExporterUtils::SetData(std::vector<T> values,
-                                      const std::vector<double> &boundaries,
-                                      const std::vector<uint64_t> &counts,
-                                      const metric_sdk::PointAttributes &labels,
-                                      std::chrono::nanoseconds time,
-                                      prometheus_client::MetricFamily *metric_family)
+void PrometheusExporterUtils::SetData(
+    std::vector<T> values,
+    const std::vector<double> &boundaries,
+    const std::vector<uint64_t> &counts,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope,
+    std::chrono::nanoseconds time,
+    prometheus_client::MetricFamily *metric_family)
 {
   metric_family->metric.emplace_back();
   prometheus_client::ClientMetric &metric = metric_family->metric.back();
-  SetMetricBasic(metric, time, labels);
+  SetMetricBasic(metric, time, labels, scope);
   SetValue(values, boundaries, counts, &metric);
 }
 
 /**
  * Set time and labels to metric data
  */
-void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &metric,
-                                             std::chrono::nanoseconds time,
-                                             const metric_sdk::PointAttributes &labels)
+void PrometheusExporterUtils::SetMetricBasic(
+    prometheus_client::ClientMetric &metric,
+    std::chrono::nanoseconds time,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope)
 {
   metric.timestamp_ms = time.count() / 1000000;
 
-  // auto label_pairs = ParseLabel(labels);
   if (!labels.empty())
   {
-    metric.label.resize(labels.size());
-    size_t i = 0;
+    metric.label.reserve(labels.size() + 2);
     for (auto const &label : labels)
     {
-      auto sanitized          = SanitizeNames(label.first);
-      metric.label[i].name    = sanitized;
-      metric.label[i++].value = AttributeValueToString(label.second);
+      prometheus_client::ClientMetric::Label sanitizedLabel;
+      sanitizedLabel.name  = SanitizeNames(label.first);
+      sanitizedLabel.value = AttributeValueToString(label.second);
+      metric.label.push_back(std::move(sanitizedLabel));
```

Comment:
Here and below consider:

```cc
metric.label.emplace_back();
metric.label.back().name = ...;
metric.label.back().value = ...;
```

Although I think I saw a helper for this in #2301

Reply:
done